enumerable.rb

```diff
@@ -0,0 +1,131 @@
+# module for some enumerable methods
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    new_array = []
+    i = 0
+    while i < size
+      new_array << yield(self[i])
+      i += 1
+    end
+    new_array
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    new_array2 = []
+    i = 0
+    while i < size
+      new_array2 << yield(self[i], i)
+      i += 1
+    end
+    new_array2
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    new_array3 = []
+    i = 0
+    while i < size
+      new_array3 << self[i] if yield(self[i]) == true
+      i += 1
+    end
+    new_array3
+  end
+
+  def my_all?
+    i = 0
+    while i < size
+      return false if yield(self[i]) == false || yield(self[i]).nil?
+
+      i += 1
+    end
+    true
+  end
+
+  def my_any?
+    contador = 0
+    i = 0
+    while i < size
+      contador += 1 if yield(self[i]) == true
+      i += 1
+    end
+    return false if contador.zero?
+
+    true
+  end
+
+  def my_none?
+    i = 0
+    while i < size
+      return false if yield(self[i]) == true
+
+      i += 1
+    end
+    true
+  end
+
+  def my_count
+    contador = 0
+    i = 0
+    while i < size
+      contador += 1 if yield(self[i])
+
+      i += 1
+    end
+    contador
+  end
+
+  def my_inject(start = 0)
+    return to_enum unless block_given?
+
+    result = start
+    i = 0
+    while i < size
+      result = yield(result, (self[i]))
+      i += 1
+    end
+    result
+  end
```

Comment:
- [ ] combines all elements of enum by applying a binary operation, specified by a block
- [ ] when a symbol is specified combines each element of the collection by applying the symbol as a named method